porch/controllers/remoterootsyncsets/pkg/controllers/remoterootsyncset/remoterootsync_controller.go

```diff
@@ -313,17 +336,94 @@ func (r *RemoteRootSyncSetReconciler) BuildObjectsToApply(ctx context.Context, s
 	return applyables, nil
 }
 
+func (r *RemoteRootSyncSetReconciler) buildObjectsToApplyFromPackage(ctx context.Context, subject *api.RemoteRootSyncSet) ([]applyset.ApplyableObject, error) {
+	packageName := subject.GetSpec().GetTemplate().GetPackageRef().GetName()
+	if packageName == "" {
+		return nil, fmt.Errorf("spec.template.packageRef.name is not set")
+	}
+
+	ns := subject.GetNamespace()
+
+	var packageRevisions porchapi.PackageRevisionList
+	labelFilter := client.MatchingLabels{
+		"kpt.dev/latest-revision": "true",
```

Comment:
Did this in #3734 as well.  If we want this to be comparable, we should probably make it an int.  We could also consider a "prior-revision" label, which might be easier to consume in practice.  But anyway, we can compare using revision (string) for now, thanks for the heads-up.

Reply:
The idea is to make a couple of functions available through the Porch API that allows users to send in a slice of PackageRevision (and/or Unstructured) resources, and we will return the latest one based on our rules for the lifecycle and the revision.

Reply:
That sounds good, though the simpler we can make the API, the more generally reusable/composable etc it will be.  But yes, if we write the library, we might see opportunities to simplify the underlying API!